### DISCUSSION TITLE
what piper has to do with espeak-ng?

### DISCUSSION THREAD

**Discussion by omega3** [rhasspy/piper#390] [Q&A] [answered]
Please explain what piper has to do with espeak-ng (on Linux). I see some threads like this [one](https://github.com/rhasspy/piper/issues/73) or [here](https://github.com/espeak-ng/espeak-ng/issues/1851) where Piper is mentioned. I try to make Piper work with speech-dispatcher and read ebooks in Calibre viewer. While with espeak is chosen everything works out of the box (but voice of course it terrible).

Is there any way to make espeak-ng or festival use Piper voices?

**Comment by synesthesiam** [comment, marked as answer]
espeak-ng has a special mode where it doesn't speak the text, but instead outputs phonemes. These phonemes are fundamental sound units of human languages, written using the International Phonetic Alphabet (IPA).

Turning text into phonemes is hard, so Piper outsources it to espeak-ng (for now). The rest of Piper is completely different from espeak-ng and festival. Those programs paste together small snippets of sounds to speak. Piper uses a machine learning model that generates audio from phonemes using a neural network. These two approaches (pasting snippets and neural networks) are not compatible, even though they both use phonemes underneath.
